src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientStopSpec.cs

```diff
@@ -0,0 +1,160 @@
+
+        protected ClusterClientStopSpec() : this(new ClusterClientStopSpecConfig())
+        {
+        }
+
+        protected ClusterClientStopSpec(ClusterClientStopSpecConfig config) : base(config)
+        {
+            _config = config;
+        }
+
+        protected override int InitialParticipantsValueFactory => 3;
+
+        private void Join(RoleName from, RoleName to)
+        {
+            RunOn(() =>
+            {
+                Cluster.Join(Node(to).Address);
+                ClusterClientReceptionist.Get(Sys);
+            }, from);
+            EnterBarrier(from.Name + "-joined");
+        }
+
+        private void AwaitCount(int expected)
+        {
+            AwaitAssert(() =>
+            {
+                DistributedPubSub.Get(Sys).Mediator.Tell(Count.Instance);
+                ExpectMsg<int>().Should().Be(expected);
+            });
+        }
+
+        private ImmutableHashSet<ActorPath> InitialContacts
+        {
+            get
+            {
+                return ImmutableHashSet
+                    .Create(_config.First, _config.Second)
+                    .Select(r => Node(r) / "system" / "receptionist")
+                    .ToImmutableHashSet();
+            }
+        }
+
+        [MultiNodeFact]
+        public void ClusterClientStopSpecs()
+        {
+            ClusterClient_must_startup_cluster();
+            ClusterClient_must_stop_if_reestablish_fails_for_too_long_time();
+        }
+
+        public void ClusterClient_must_startup_cluster()
+        {
+            Within(30.Seconds(), () =>
+            {
+                Join(_config.First, _config.First);
+                Join(_config.Second, _config.First);
+
+                RunOn(() =>
+                {
+                    var service = Sys.ActorOf(Props.Create(() => new ClusterClientStopSpecConfig.Service()), "testService");
+                    ClusterClientReceptionist.Get(Sys).RegisterService(service);
+                }, _config.First);
+
+                RunOn(() =>
+                {
+                    AwaitCount(1);
+                }, _config.First, _config.Second);
+
+                EnterBarrier("cluster-started");
+            });
+        }
+
+        public void ClusterClient_must_stop_if_reestablish_fails_for_too_long_time()
+        {
+            Within(20.Seconds(), () =>
+            {
+                RunOn(() =>
+                {
+                    var c = Sys.ActorOf(ClusterClient.Props(ClusterClientSettings.Create(Sys).WithInitialContacts(InitialContacts)), "client1");
+                    c.Tell(new ClusterClient.Send("/user/testService", "hello", localAffinity: true));
+                    ExpectMsg<string>().Should().Be("hello");
+                    EnterBarrier("was-in-contact");
+                    Watch(c);
+
+                    ExpectTerminated(c, 30.Seconds());
+                    EventFilter.Warning(start: "Receptionist reconnect not successful within").ExpectOne(() => { });
```

Comment:
The test fails here. It is weird, because I can see this warning in my log file
```
[WARNING][04.08.2016 12:53:49][Thread 0020][[akka://MultiNodeClusterSpec/user/client1#914935706]] Receptionist reconnect not successful within 00:00:03 stopping cluster client
```